src/linux/component.rs

```diff
@@ -9,152 +14,392 @@ use std::path::{Path, PathBuf};
-                    });
-                    let crit = get_file_line(p_crit.as_path(), 10).map(|crit| {
-                        crit.replace('\n', "").parse::<f32>().unwrap_or(100_000f32) / 1000f32
-                    });
-                    components.push(Component::new(label, p_input, max, crit));
-                }
+    }
+}
+
+/// Check given `item` dispatch to read the right `file` with the right parsing and store data in
+/// given `component`. `id` is provided for `label` creation.
+fn fill_component(component: &mut Component, item: &str, folder: &Path, file: &str) {
+    let hwmon_file = folder.join(file);
+    match item {
+        "type" => {
+            component.sensor_type =
+                read_number_from_file::<u8>(&hwmon_file).map(|n| TermalSensorType::from(n))
+        }
+        "input" => {
+            let temperature = get_temperature_from_file(&hwmon_file);
+            component.input_file = hwmon_file;
+            component.temperature = temperature;
+            // Max computed value taken form current temperatue if not possible to
+            // get it from `highest`.
+            if component.max.is_none() {
+                component.max = temperature;
             }
         }
+        "label" => component.label = get_file_line(&hwmon_file, 10).unwrap_or(String::new()),
+        "highest" => {
+            component.max = get_temperature_from_file(&hwmon_file).or(component.temperature);
+            component.highest_file = Some(hwmon_file);
+        }
+        "max" => component.threshold_max = get_temperature_from_file(&hwmon_file),
+        "min" => component.threshold_min = get_temperature_from_file(&hwmon_file),
+        "crit" => component.threshold_critical = get_temperature_from_file(&hwmon_file),
+        // Raspberry_pi weird comportement.
+        // TODO: Test it on a raspberry pi.
+        "" if component.label.is_empty() => {
+            component.label = format!("Component");
+            sysinfo_debug!("Raspberry pi empty label encountered: {:?}"
+            "\n Warning component may be partially constructed.", component)
+        }
+        // TODO maybe it's good to trace for being future aware of a change in hwmon temp?
+        _ => {}
     }
 }
 
+/// Read out `hwmon` info (hardware monitor) from `folder` [Path]
+/// for value path to check at refresh and files containing `max`,
+/// `critical value and `label` and store this info in `components`.
+///
+/// What is read:
+///
+/// - Optionnal: sensor label, in the general case content of `tempN_label`
+///   see below for special cases
+/// - Optionnal: max value defined in `tempN_max`
+/// - Optionnal: critical value defined in `tempN_crit`
+///
+/// Where `N` is a u32 associated to a sensor like `temp1_max`, `temp1_input`.
+///
+/// ## Special case: Disk drive with `drivetemp` module
+///
+/// For some crazy rational, hardware monitoring exposed by `drivetemp` has no label so the `label of a `Component`
```

Comment:
```suggestion
/// For some crazy rational, hardware monitoring exposed by `drivetemp` has no label so the `label` of a `Component`
```

Reply:
finnaly droped since nothing is special cased.